src/dashboard/alerts/alert.py

```diff
@@ -64,20 +40,24 @@ def __init__(self, pkt: Packet, alert_description="", alert_type=ALERT_TYPE.UNKN
         self.type = str(alert_type)
         self.severity = int(alert_severity)
 
-        if is_destination:
-            if "IP" in pkt:
-                self.device_ip = pkt["IP"].dst
-            if "Ethernet" in pkt:
-                self.device_mac = pkt["Ethernet"].dst
-        else:
-            if "IP" in pkt:
-                self.device_ip = pkt["IP"].src
-            else:
-                self.device_ip = "[Layer 2]"
-            if "Ethernet" in pkt:
-                self.device_mac = pkt["Ethernet"].src
+        if pkt:
+            if is_destination:
+                if "IP" in pkt:
+                    self.device_ip = pkt["IP"].dst
+                if "Ethernet" in pkt:
+                    self.device_mac = pkt["Ethernet"].dst
             else:
-                self.device_mac = "[Unknown]"
+                if "IP" in pkt:
+                    self.device_ip = pkt["IP"].src
+                else:
+                    self.device_ip = "[Layer 2]"
+                if "Ethernet" in pkt:
+                    self.device_mac = pkt["Ethernet"].src
+                else:
```

Comment:
Can you explain why these checks are being done twice in this block of code? They can probably be refactored to simplify the code and make it more readable as well.

Reply:
I can organize it to remove the inner else statements and just assume that every packet will have an _Ethernet_ segment (Hopefully that won't cause a crash, but honestly if we do get a packet without an Ethernet segment I'll be kinda amazed), but its still 2 layers deep of a binary decision tree so there are 4 outcomes and I can't really reduce the amount of checks for IP because they have independent outcomes.

Reply:
Maybe just add a try/catch for edge cases

Reply:
Can do. The only edge cases it could get would be malformed test packets from within our own tests, but I can add it